doc: add NFC documentation (#176)
nrfconnect/sdk-nrfxlib: nfc/doc/type_2_tag.rst

.. _type_2_tag:

Type 2 Tag
##########

> 

Reviewer: Maybe we should refer to the NFC Forum _Type 2 Tag Technical Specification Version 1.0_ (like with do in case of T4T)? A downside is we'd need to remember to keep this info up-to-date.

> The Type 4 Tag implementation is based on the NFC Forum document *Type 2 Tag Technical Specification Version 1.0*.

Author: Darn, I knew copying the sentence over was a bad idea. ;)
Fixed.

A Type 2 Tag can be read and re-written, and the memory of the tag can be write protected.
The Type 2 Tag library implements a Type 2 Tag in read-only state.
Read only means that the memory of the tag is write protected and cannot be erased (formatted) or re-written by the polling device.

> If you use the supplied library, you do not need to care about the actual :ref:`t2t_memory_layout` that is used to implement the tag.

Reviewer 2: ```suggestion
If you use the supplied library, you do not need to modify anything in the :ref:`t2t_memory_layout` that is used to implement the tag.
```

Author: Updating to "If you use the supplied library, you do not need to **know** about the actual :ref:`t2t_memory_layout` that is used to implement the tag.", because you wouldn't modify anything here anyway, but the idea is that you can skip this section.

     - Description
     - Value
   * - 1
     - Code
     - 0x30
   * - 2
     - Number of the 4-byte data block to read from the tag data
     - 0x00 - 0xFF

The library can send either a READ response (success) or a NACK response (failure):

* A READ response contains the content of 4 data blocks starting with the requested data block (16 bytes).
* A NACK response contains the error code 0x0, 0x1, 0x4, or 0x5 (4 bits).

.. _t2t_programming:

Programming a tag
*****************

To program a tag, complete the following steps:

1. Implement a callback function that handles events from the Type 2 Tag library and register it:

   .. code-block:: c

      int err;
      /* Callback for NFC events */
      static void nfc_callback(void * context,
                               enum nfc_t4t_event event,
                               const u8_t * data,
                               size_t data_length,
                               u32_t flags)
      {
      ...
      }
      /* Set up NFC and register application callback for NFC events. */
      err = nfc_t2t_setup(nfc_callback, NULL);
      if (err) {
		printk("Cannot setup NFC T2T library!\n");
		return err;
      }

>    The NFCT interrupt handler runs at priority level APP_LOW.

Reviewer: There is no _APP_LOW_ in NCS, there is a Kconfig symbol in NFC Platform in NCS. So, I think either we should remove this information from here, or just say it's configurable by the NFC Platform. I wouldn't link to the exact Kconfig setting as that is part of NCS and theoretically other users of this library can set their own IRQ priority in their own way, depending of the runtime environment.

Author: removed
